tests/test_cli_replace_modal_subgraph.py

```diff
@@ -0,0 +1,195 @@
+        modes="bike",
+        increase_capacity=True,
+    )
+
+    expected_new_link_id = sub_network["expected_bike_link_id"]
+    assert output_network.has_link(
+        expected_new_link_id
+    ), f"Link {expected_new_link_id} is missing from the output network"
+    assert output_network.link(expected_new_link_id)["modes"] == {
+        "bike"
+    }, f"Link {expected_new_link_id} did not have the correct modes set"
+
+
+def test_retains_link_data_from_subgraph_network(tmpdir, network, sub_network):
+    original_link_data = sub_network["network"].link(sub_network["bike_link_id"])
+
+    output_network = cli._replace_modal_subgraph(
+        path_to_network=network["path"],
+        projection="epsg:27700",
+        output_dir=tmpdir,
+        path_to_subgraph=sub_network["path"],
+        modes="bike",
+        increase_capacity=True,
+    )
+
+    output_data = output_network.link(sub_network["expected_bike_link_id"])
+    keys_to_ignore = {"from", "s2_from", "to", "s2_to", "id", "modes", "capacity"}
+    assert {k: v for k, v in output_data.items() if k not in keys_to_ignore} == {
+        k: v for k, v in original_link_data.items() if k not in keys_to_ignore
+    }, "Data for the added link does not match the original data of the link"
+
+
+def test_increases_capacity_for_added_links_only(tmpdir, network, sub_network):
+    original_net_capacity = {
+        link_id: data["capacity"] for link_id, data in network["network"].links()
+    }
+
+    output_network = cli._replace_modal_subgraph(
+        path_to_network=network["path"],
+        projection="epsg:27700",
+        output_dir=tmpdir,
+        path_to_subgraph=sub_network["path"],
+        modes="bike",
+        increase_capacity=True,
+    )
+
+    expected_new_link_id = sub_network["expected_bike_link_id"]
+    assert output_network.has_link(
+        expected_new_link_id
+    ), f"Link {expected_new_link_id} is missing from the output network"
+    assert (
+        output_network.link(expected_new_link_id)["capacity"] == 9999
+    ), f"Link {expected_new_link_id} did not have capacity increased"
+
+    link_id = network["link_id"]
+    assert output_network.has_link(link_id), f"Link {link_id} is missing from the output network"
+    assert (
+        output_network.link(link_id)["capacity"] == original_net_capacity[link_id]
+    ), f"Link {link_id} did not retain th original capacity value"
+
+
+def test_does_not_increase_capacity_if_not_requested(tmpdir, network, sub_network):
+    original_subnet_capacity = {
+        link_id: data["capacity"] for link_id, data in sub_network["network"].links()
+    }
+
+    output_network = cli._replace_modal_subgraph(
+        path_to_network=network["path"],
+        projection="epsg:27700",
+        output_dir=tmpdir,
+        path_to_subgraph=sub_network["path"],
+        modes="bike",
+        increase_capacity=False,
+    )
+
+    for link_id, old_capacity in original_subnet_capacity.items():
```

Comment:
Again, it might help to extract the comparison into a helper function with a name that makes what we're asserting obvious but hides the details of how we perform that comparison.